File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/sync/CollectServiceFilesForSync.java
package com.lasthopesoftware.bluewater.client.stored.library.sync;

import com.lasthopesoftware.bluewater.client.browsing.items.media.files.ServiceFile;
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId;
import com.namehillsoftware.handoff.promises.Promise;

import java.util.Collection;

public interface CollectServiceFilesForSync {
	Promise<Collection<ServiceFile>> promiseServiceFilesToSync(LibraryId libraryId);
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/stored/library/sync/CollectServiceFilesForSync.kt
package com.lasthopesoftware.bluewater.client.stored.library.sync

import com.lasthopesoftware.bluewater.client.browsing.files.ServiceFile
import com.lasthopesoftware.bluewater.client.browsing.library.repository.LibraryId
import com.namehillsoftware.handoff.promises.Promise

interface CollectServiceFilesForSync {
	fun promiseServiceFilesToSync(libraryId: LibraryId): Promise<Iterable<ServiceFile>>
}
